docs/includes/extracts-bucket-option.yaml

```diff
@@ -1,3 +1,11 @@
+ref: bucket-option-codec
+content: |
+  The :doc:`codec </tutorial/codecs>` to use for encoding or decoding documents.
+  This option is mutually exclusive with the ``typeMap`` option.
+
+  Defaults to the bucket's codec. Inheritance for a default ``codec`` option
```

Comment:
> This means if both the codec and the typeMap options are set in the bucket, we take the ~~bucket~~ codec.

Corrected what I assume is a typo.

> And if the codec is set to null in this method call, we use the typeMap from the Bucket?

That is correct.

The PHPLIB docs have never discussed passing `null` for options. `null` values are typically ignored when we process options in Operation classes due to the use of `isset()`.

If the `codec` option is `null`, the `Bucket::find()` and `findOne()` methods will still send it along to the CollectionWrapper since they use `array_key_exists()`; however, when it is ultimately received by the files collection's `Collection::find()` or `findOne()` method, the `null` value will be ignored based on the logic in `Collection::inheritCodecOrTypeMap()`.